contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordENoteContract.kt

```diff
@@ -16,11 +17,13 @@ import tech.figure.servicing.v1beta1.LoanStateOuterClass.ServicingData
 @Participants(roles = [PartyType.OWNER])
 @ScopeSpecification(["tech.figure.loan"])
 open class RecordENoteContract(
+    @Record(name = LoanScopeFacts.eNote, optional = true) val existingENote: ENote?,
```

Comment:
Im a bit confused why we'd have an existing enote and need to call recordEnoteContract? It seems we've made this contract for recording enotes OR servicing data? Should we not just have a contract for that instead?

Reply:
I think the original desire was for the `recordServicingData` function to be added to this contract as a convenience, but @dkneisly-figure or @mnguyen-figure may be able to speak to that further. I simply added the `@SkipIfRecordExists` to `recordENote` to prevent this contract from being able to further update the eNote after initializing it on the scope, and said annotation requires the record to be an optional value in the contract's constructor.